src/ServiceControl/Infrastructure/MetadataExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace ServiceControl
+{
+    using System;
+    using System.Collections.Generic;
+
+    static class MetadataExtensions
+    {
+        public static T MaybeGet<T>(this IDictionary<string, object> metadata, string key)
```

Comment:
How about `GetOrDefault` ?

Reply:
I would prefer GetOrDefault as well

Reply:
What about the `MaybeGetAsString()`? `GetOrNullAsString()`?

Reply:
`GetAsStringOrNull`